### PR TITLE
Heart eater wizard perk no longer works activates on organs that haven't been used before

### LINKED ISSUE #85364: heart eater oversight
## Issue Summary
The heart eater perk for a wizard allows them to get damage resistance when eating hearts.
The issue is it also works for eating cybernetic hearts when they're deep-fried.

## Round ID:
local

## Reproduction:
1. pick heart eater perk as a wizard
2. deep-fry a cybernetic heart
3. eat it
4. profit

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Closes #85364
Organs now mark themselves after being removed from someone, which is that the perk checks for

## Changelog
:cl:
fix: Heart eater wizard perk no longer works activates on organs that haven't been used before
/:cl:

### CODE REVIEW

File: code/datums/components/heart_eater.dm [hunk 1]
```diff
@@ -65,6 +65,8 @@
 	if(!istype(what_we_ate, /obj/item/organ/internal/heart))
 		return
 	var/obj/item/organ/internal/heart/we_ate_heart = what_we_ate
+	if (!HAS_TRAIT(we_ate_heart, TRAIT_USED_ORGAN))
```
Comment (Jacquerel) [inline code/datums/components/heart_eater.dm:68]: Preferably give some kind of balloon alert feedback here ("no life energy!"?)

Comment (Jacquerel) [reply code/datums/components/heart_eater.dm:68]: Actually `to_chat` is probably better given that this isn't blocking; they will still eat the heart.

**Comment by carlarctg** [comment]
if you cared about soul you'd keep this but add a limit to the amount, so that you cant mass lathe the organs

**Comment by Jacquerel** [comment]
> if you cared about soul you'd keep this but add a limit to the amount, so that you cant mass lathe the organs

This is more soulful than the feature already was already, by checking if it was ever inside a living person.
"You can eat printed organs but only some" isn't soulful at all it's just powergamer enabling.

**Comment by zxaber** [comment]
Alternatively, this still allows you to eat them as long as you insert and then remove the organ, which implies some sort of terrifying imagery about needing to season the heart inside a chest cavity first.

**Pull request merged**